src/platform_impl/macos/view.rs

```diff
@@ -762,9 +776,10 @@ extern "C" fn flags_changed(this: &Object, _sel: Sel, event: id) {
             }));
         }
 
-        AppState::queue_event(EventWrapper::StaticEvent(Event::DeviceEvent {
-            device_id: DEVICE_ID,
-            event: DeviceEvent::ModifiersChanged(state.modifiers),
+        AppState::queue_event(EventWrapper::StaticEvent(Event::WindowEvent {
+            // TODO Maybe memoize get_window_id if it's safe to reuse?
+            window_id: WindowId(get_window_id(state.ns_window)),
```

Comment:
The window ID of the state isn't going to change. If it is extracted it should probably be used in the `for event in events` loop too. Would probably be a good change.